### PR TITLE
Support for multi-speaker 

### PR DESCRIPTION
**Pull request by r9y9**
continued from #6

### CODE REVIEW

**Comment by r9y9** [comment]
For the recoder, at https://github.com/r9y9/deepvoice3_pytorch/pull/10/commits/26ab716648e6dc2d51a66122cb0ce5ca9bf7a342, I think I can get a reasonable speech quality as noted in #1. Things I have found so far to make multi-speaker model actually work are:

- Apply dropout for speaker embedding (I first expanded speaker embedding to all time steps and then applied dropout. It seems work, but it's worth trying to first apply dropout and then expand
- Use single attention layer rather than multiple ones, otherwise attention often fail at incremental decoding

**Comment by r9y9** [comment]
Cherry-picked a few commits to master.

**Comment by r9y9** [comment]
Okay, I think I have finished multi-speaker support. After merging this PR, I will upload a pre-trained model and audio samples.
